src/core/include/units/bits/algorithm.h

```diff
@@ -81,7 +81,7 @@ constexpr auto lexicographical_compare_three_way(I1 f1, I1 l1, I2 f2, I2 l2, Cmp
   bool exhaust1 = (f1 == l1);
   bool exhaust2 = (f2 == l2);
   for (; !exhaust1 && !exhaust2; exhaust1 = (++f1 == l1), exhaust2 = (++f2 == l2))
-    if (auto c = comp(*f1, *f2); c != 0) return c;
+    if (auto c = comp(*f1, *f2); c != nullptr) return c;
```

Comment:
This is UB: https://eel.is/c++draft/cmp.categories#pre-3.

Reply:
Done.